aggregator/src/config.rs

```diff
@@ -43,12 +43,22 @@ pub struct CommonConfig {
     /// Address to serve HTTP health check requests on.
     #[serde(default = "default_health_check_listen_address")]
     pub health_check_listen_address: SocketAddr,
+
+    /// The maximum number of times a transaction can be retried. The intent is to guard against bugs
+    /// that induce infinite retries. It should be set to a reasonably high limit to prevent legitimate
+    /// work from being cancelled.
+    #[serde(default = "default_max_transaction_retries")]
+    pub max_transaction_retries: u64,
 }
 
 fn default_health_check_listen_address() -> SocketAddr {
     SocketAddr::new(IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)), 9001)
 }
 
+pub fn default_max_transaction_retries() -> u64 {
+    1000
```

Comment:
Why 1000? Is this informed by retry counts we've observed in production?

Reply:
Yes, insofar as I've not seen retry counts that approach this number--the maximum retry count I've seen in production is about 20, corresponding to normal serialization retries.

I target this number so as not to abort transactions that are retrying due to concurrency errors, rather only abort in case of application-driven retry bugs, so I leave considerable safety margin.